staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -136,10 +137,13 @@ func NestedNumberAsFloat64(obj map[string]interface{}, fields ...string) (float6
 	}
 	switch x := val.(type) {
 	case int64:
-		if x != int64(float64(x)) {
+		var bf big.Float
+		bf.SetInt64(x)
```

Comment:
does Float#SetInt64 remember that rounding occurred?

wouldn't it make more sense to use `f, accuracy := big.NewInt(x).Float64()`?